pkg/kubectl/cmd/create/create_clusterrole.go

```diff
@@ -103,10 +103,6 @@ func (c *CreateClusterRoleOptions) Complete(f cmdutil.Factory, cmd *cobra.Comman
 }
 
 func (c *CreateClusterRoleOptions) Validate() error {
-	if c.Name == "" {
```

Comment:
Not right here.

Here `NameFromCommandArgs` is not called in `Complete`, which means `c.Name` could be an empty string. That's why current check is needed.

Reply:
@dixudx `NameFromCommandArgs()` is called via `c.CreateRoleOptions.Complete()`.

https://github.com/kubernetes/kubernetes/blob/b2fe2a0a6d1f72ac9041ca9f9d0dac2e5d8a57c4/pkg/kubectl/cmd/create/create_clusterrole.go#L102

But we need the double check in `create_clsuterrole.go`?

Reply:
OK, reverted the name arg check for clusterrole.